packages/js/uri-resolvers/src/helpers/RetryResolver.ts

```diff
@@ -0,0 +1,113 @@
+import { UriResolverLike } from "./UriResolverLike";
+import { UriResolver } from "./UriResolver";
+import { UriResolutionResult } from "./UriResolutionResult";
+
+import {
+  CoreClient,
+  IUriResolutionContext,
+  IUriResolver,
+  Result,
+  Uri,
+  UriPackageOrWrapper,
+} from "@polywrap/core-js";
+
+/** configure how additional resolution attempts are handled after an initial resolution attempt fails */
+export type RetryOptions = {
+  /** the number of additional resolution attempts */
+  retries: number;
+  /** the duration (in ms) to pause between attempts */
+  interval: number;
+};
+
+/** A map or uri or authority strings to retry options */
+export type RetryResolverOptions = {
+  [uriOrAuthority: string]: RetryOptions;
+};
+
+export class RetryResolver<TError = undefined> implements IUriResolver<TError> {
+  constructor(
+    private resolver: IUriResolver<TError>,
+    private options: RetryResolverOptions
+  ) {}
+
+  static from<TResolverError = unknown>(
+    resolver: UriResolverLike,
+    options: RetryResolverOptions
+  ): RetryResolver<TResolverError> {
+    return new RetryResolver(
+      UriResolver.from<TResolverError>(resolver),
+      options
+    );
+  }
+
+  async tryResolveUri(
+    uri: Uri,
+    client: CoreClient,
+    resolutionContext: IUriResolutionContext
+  ): Promise<Result<UriPackageOrWrapper, TError>> {
+    const subContext = resolutionContext.createSubHistoryContext();
+
+    const result = await this.resolver.tryResolveUri(uri, client, subContext);
+
+    const isChange = !(
+      result.ok &&
```

Comment:
So this will retry if (and only if) a URI is not found?
Is that what we want? I thought we want to retry in case of an error (like a request timeout=

Reply:
If there is an error, such as a timeout, it is currently returned as no change with `result.ok === true`